sdk/lib/core/bool.dart

```diff
@@ -19,6 +19,8 @@ class bool {
   /// In all other cases, including when there is no declaration for `name`,
   /// the result is the [defaultValue].
   ///
+  /// The value must be assigned to a constant.
```

Comment:
That's actually not correct. You can do:
```dart
 var davar = const String.fromEnvironment("daname", defaultValue: "dafault");
```

What is true is that the constructor invocation must be constant when used on the web (perhaps other ahead-of-time compiled targets too).
That's a restriction which was added by the web compilers (and semi-documented in line 44 below).
We should figure out whether we want that to be a general restriction (affecting the VM too), and then document whichever choice we make.

Reply:
Oh, cool, I didn’t know that, thanks for the information, so, do you think that still worth updating the docs with this information?

Reply:
Something like The .fromEnvironment() constructors must be constant.

Reply:
@lrhn I updated the docs, do you think that's good enough now?

Reply:
It's correct and complete. 
I'm a little worried that it lacks context (AoT is an undefined abbreviation, but explaining it also seems like overkill. So, a matter of trade-off between readability and detail.) 
The documentation should apply to all four `Environment` functions.

I'll try writing something up.